src/data/clean_data.py

```diff
@@ -0,0 +1,245 @@
+import pandas as pd
+import fnmatch
+
+# Imports for ngrams()
+import re
+from ftfy import fix_text
+
+# Imports for awesome_cossim_top()
+import numpy as np
+from scipy.sparse import csr_matrix, isspmatrix_csr
+import sparse_dot_topn.sparse_dot_topn as ct
+
+# Import for vectorize
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+def _csv_to_df(path: str):
+    df = pd.read_csv(f"data/raw_data/{path}", index_col=0)
+    df.fillna(value="", inplace=True)
+    return df
+    
+        
+
+def _get_df_columns(df):
+    column_list = df.columns.values.tolist()
+    return column_list
+
+
+class DataCleaner:
+    """
+    Instantiated with a dataframe its column names, and size of lowest_similarity and ngram_size. 
+    DataCleaner methods to provide a smart deduplication of column values in a dataframe.
+    Utilizes a form of Term Frequency, Inverse Document Frequency to determine similarity between values.
+
+
+    Parameters:
+        path: Pandas.DataFrame.
+            Use _csv_to_df to provide df from csv.
+        
+        lowest_similarity: float
+            Lowest similarity percentage of column values to combine.
+                i.e. 0.9 = words with 90% or more similarity are combined.
+
+        column_name: str
+            A column name from path. Use _get_df_columns to get a list of columns.
+
+        ngram_size: int
+            Size of string chunks used to assess similarity between two values.
+            3 is normally best but values between 2 and 5 can work.
+
+    Returns:
+        A Pandas DataFrame with similar values of column_name combined.
+
+    """
+
+    def __init__(self, path: pd.DataFrame, lowest_similarity: float, column_name: str, ngram_size: int):
+        self.df = path
+        self.lowest_similarity = lowest_similarity
+        self.column_name = column_name
+        self.ngram_size = ngram_size
+        
+
+    def _ngrams(self, string):
+        """
+        Takes input `string`, removes unwanted chars and returns a list of slices of input `string`.
```

Comment:
That's a lot of (useful) code below. Is there something more informative than "unwanted" we could use to explain what's getting stripped?

Reply:
unwanted really feels unwanted right now